### PR TITLE
Browser extension: make "View file (base) + (head)" one button

### PR DESCRIPTION
**Pull request by attfarhan**
Part of https://github.com/sourcegraph/sourcegraph/issues/1178.

Changes the "View file (base/head)" to be a single  "View file diff" button which links to the corresponding file in the Sourcegraph diff.

### CODE REVIEW

File: client/browser/src/shared/backend/diffInternalIDs.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        base: string | null
+        head: string | null
+        first?: number
+    }): Observable<GQL.IFileDiffConnection> =>
+        queryGraphQL({
+            ctx: { repoKey: '', isRepoSpecific: false },
+            request: `
+            query RepositoryComparisonDiff($repo: String!, $base: String, $head: String, $first: Int) {
+                repository(name: $repo) {
+                    ... on Repository {
```
Comment (felixfbecker) [inline client/browser/src/shared/backend/diffInternalIDs.tsx]: What else can `repository` return?

File: client/browser/src/shared/components/OpenDiffOnSourcegraph.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+    private getOpenInSourcegraphUrl(props: OpenDiffInSourcegraphProps): string {
+        const baseUrl = repoUrlCache[props.repoName] || sourcegraphUrl
+        const url = `${baseUrl}/${props.repoName}`
+        const urlToCommit = `${url}/-/compare/${props.commit.baseRev}...${
+            props.commit.headRev
+        }?utm_source=${getPlatformName()}`
+
+        if (this.state.fileDiff) {
+            if (this.state.fileDiff.nodes.length > 0) {
```
Comment (felixfbecker) [inline client/browser/src/shared/components/OpenDiffOnSourcegraph.tsx]: combine ifs

File: client/browser/src/shared/backend/diffInternalIDs.tsx (hunk 1 above)
Comment (attfarhan) [reply client/browser/src/shared/backend/diffInternalIDs.tsx]: Nothing else, fixed. Thanks!

**Comment by attfarhan** [comment]
@felixfbecker look good to you? (Other than merge conflict. Will fix when I wake up.)

File: client/browser/src/shared/components/OpenDiffOnSourcegraph.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+import { getPlatformName, repoUrlCache, sourcegraphUrl } from '../util/context'
+import { Button } from './Button'
+
+export interface Props {
+    openProps: OpenDiffInSourcegraphProps
+    style?: React.CSSProperties
+    iconStyle?: React.CSSProperties
+    className?: string
+    ariaLabel?: string
+    onClick?: (e: any) => void
```
Comment (felixfbecker) [inline client/browser/src/shared/components/OpenDiffOnSourcegraph.tsx]: `e` shouldn't be `any`

**Pull request merged**